content/docs/intro/concepts/_index.md

```diff
@@ -9,49 +9,292 @@ menu:
+)
+
+func main() {
+    pulumi.Run(func(ctx *pulumi.Context) error {
+        group, err := ec2.NewSecurityGroup(ctx, "web-sg", &ec2.SecurityGroupArgs{
+            Description: pulumi.String("Enable HTTP access"),
+            Ingress: ec2.SecurityGroupIngressArray{
+                ec2.SecurityGroupIngressArgs{
+                    Protocol:   pulumi.String("tcp"),
+                    FromPort:   pulumi.Int(80),
+                    ToPort:     pulumi.Int(80),
+                    CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+                },
+            },
+        })
+        if err != nil {
+            return err
+        }
+        server, err := ec2.NewInstance(ctx, "web-server", &ec2.InstanceArgs{
+            Ami:            pulumi.String("ami-6869aa05"),
+            InstanceType:   pulumi.String("t2.micro"),
+            SecurityGroups: pulumi.StringArray{group.Name},
+        })
+        if err != nil {
+            return err
+        }
+        return nil
+    })
+}
+```
+
+{{% /choosable %}}
+{{% choosable language csharp %}}
+
+```csharp
+using System.Threading.Tasks;
+using Pulumi;
+using Pulumi.Aws.Ec2;
+using Pulumi.Aws.Ec2.Inputs;
+
+class Program
+{
+    static Task<int> Main() => Deployment.RunAsync<MyStack>();
+}
+
+
+class MyStack : Stack
+{
+    public MyStack()
+    {
+        var group = new SecurityGroup("web-sg", new SecurityGroupArgs {
+            Description = "Enable HTTP access",
+            Ingress = {
+                new SecurityGroupIngressArgs {
+                    Protocol = "tcp",
+                    FromPort = 80,
+                    ToPort = 80,
+                    CidrBlocks = { "0.0.0.0/0" }
+                }
+            }
+        });
+        var server = new Instance("web-server", new InstanceArgs {
+            Ami = "ami-6869aa05",
+            InstanceType = "t2.micro",
+            SecurityGroups = { group.Name }
+        });
+    }
+}
+```
+
+{{% /choosable %}}
+
+{{< /chooser >}}
+
+The two resource objects, and their properties, are used by Pulumi to perform the appropriate actions on your infrastructure. For example, Pulumi understands that you would like an EC2 security group named `web-sg`, with a single ingress rule and a `t2.micro`-sized EC2 instance running AMI `ami-8689aa05` using that security group. And because of the output properties, Pulumi also understands the dependencies between resources, which maximizes parallelism and ensures correct ordering.
+
+When you update your cloud project with Pulumi, Pulumi will compute the desired state, compare it to the current infrastructure you already have (if any), show you the delta, and let you confirm and carry out the changes.
+
+If needed, you can also export resulting infrastructure values to access outside your application. For example, adding the following code to the example above exports the server's resulting IP address and DNS name to either stdout or for use by another stack:
```

Comment:
In the form this is in here - might make sense to just include this in the original example, and then talk through how this program shows off each of the concepts - including "stack outputs".